sympy/solvers/ode/tests/test_ode.py

```diff
@@ -33,6 +33,112 @@
 
 
+def _ode_solver_test(ode_examples,our_hint='default'):
+    for example in ode_examples:
+        eq = ode_examples[example]['eq']
+        sol = ode_examples[example]['sol']
+        assert our_hint in classify_ode(eq), our_hint+"Hint was not recognised for the equation "+str(eq)
+        sols = sol + [constant_renumber(root) for root in sol]
+        sols += [root.expand() for root in sol]
+        expected_checkodesol = [(True, 0) for i in range(len(sol))]
+        assert dsolve(eq,hint=our_hint).rhs in sols, "Expected solution is different from solution returned by dsolve for the equation "+str(eq)
+        assert checkodesol(eq, sol) == expected_checkodesol, "The solution returned by dsolve is not correct for the equation "+str(eq)
+
+
+ode_sol_euler_homogeneous = {
+    'euler_hom_01': {
+        'eq': Eq(-3*diff(f(x), x)*x + 2*x**2*diff(f(x), x, x), 0),
+        'sol': [C1 + C2*x**Rational(5, 2)]
+    },
+
+    'euler_hom_02': {
+        'eq': Eq(3*f(x) - 5*diff(f(x), x)*x + 2*x**2*diff(f(x), x, x), 0),
+        'sol': [C1*sqrt(x) + C2*x**3]
+    },
+
+    'euler_hom_03': {
+        'eq': Eq(4*f(x) + 5*diff(f(x), x)*x + x**2*diff(f(x), x, x), 0),
+        'sol': [(C1 + C2*log(x))/x**2]
+    },
+
+    'euler_hom_04': {
+        'eq': Eq(6*f(x) - 6*diff(f(x), x)*x + 1*x**2*diff(f(x), x, x) + x**3*diff(f(x), x, x, x), 0),
+        'sol': [C1/x**2 + C2*x + C3*x**3]
+    },
+
+    'euler_hom_05': {
+        'eq': Eq(-125*f(x) + 61*diff(f(x), x)*x - 12*x**2*diff(f(x), x, x) + x**3*diff(f(x), x, x, x), 0),
+        'sol': [x**5*(C1 + C2*log(x) + C3*log(x)**2)]
+    },
+
+    'euler_hom_06': {
+        'eq': x**2*diff(f(x), x, 2) + x*diff(f(x), x) - 9*f(x),
+        'sol': [C1*x**3 + C2*x**-3]
+    },
+
+    'euler_hom_07': {
+        'eq': sin(x)*x**2*f(x).diff(x, 2) + sin(x)*x*f(x).diff(x) + sin(x)*f(x),
+        'sol': [C1*sin(log(x)) + C2*cos(log(x))]
+    },
+}
+
+
+ode_sol_euler_undetermined_coeff = {
+    'euler_undet_01': {
+        'eq': Eq(x**2*diff(f(x), x, x) + x*diff(f(x), x), 1),
+        'sol': [C1 + C2*log(x) + log(x)**2/2]
+    },
+
+    'euler_undet_02': {
+        'eq': Eq(x**2*diff(f(x), x, x) - 2*x*diff(f(x), x) + 2*f(x), x**3),
+        'sol': [x*(C1 + C2*x + Rational(1, 2)*x**2)]
+    },
+
+    'euler_undet_03': {
+        'eq': Eq(x**2*diff(f(x), x, x) - x*diff(f(x), x) - 3*f(x), log(x)/x),
+        'sol': [(C1 + C2*x**4 - log(x)**2/8 - log(x)/16)/x]
+    },
+
+    'euler_undet_04': {
+        'eq': Eq(x**2*diff(f(x), x, x) + 3*x*diff(f(x), x) - 8*f(x), log(x)**3 - log(x)),
+        'sol': [C1/x**4 + C2*x**2 - Rational(1,8)*log(x)**3 - Rational(3,32)*log(x)**2 - Rational(1,64)*log(x) - Rational(7, 256)]
+    },
+
+    'euler_undet_05': {
+        'eq': Eq(x**3*diff(f(x), x, x, x) - 3*x**2*diff(f(x), x, x) + 6*x*diff(f(x), x) - 6*f(x), log(x)),
+        'sol': [C1*x + C2*x**2 + C3*x**3 - Rational(1, 6)*log(x) - Rational(11, 36)]
+    },
+}
+
+
+ode_sol_euler_var_para = {
+    'euler_var_01': {
```

Comment:
I think that this should have a key specifying which hint these ODEs correspond to like:
```
ode_sol_euler_var_para = {{
    'hint':'nth_euler_variation_of_parameters',
    'examples':{

    'euler_var_01': {
...
}}
```
That will be useful when running other tests on the examples outside of the standard tests here.